### PR TITLE
Added reload action to the opened files context menu.

### PR DESCRIPTION
**Pull request by martonmiklos**
When using an external CAM software it is common use case that the modified G code needs to be reloaded. 
I have added an action to the context menu to make this action more easier.

### CODE REVIEW

**Comment by martonmiklos** [comment]
I have added reload file to the context menu:
kep
And recent projects to the file menu:
kep

File: src/camotics/qt/QtWin.cpp [hunk 1]
```diff
@@ -1541,6 +1598,11 @@ void QtWin::animate() {
   }
 }
 
+void QtWin::openRecentProjectsSlot(const QString path)
+{
```
Comment (jcoffland) [inline src/camotics/qt/QtWin.cpp]: This brace should be on the line above.

File: src/camotics/qt/QtWin.cpp [hunk 2]
```diff
@@ -818,6 +826,55 @@ void QtWin::openProject(const string &_filename) {
...
+    // add the new menu item
+    QAction *action = ui->menuRecent_projects->addAction(QString::fromStdString(filename), &recentProjectsMapper, SLOT(map()));
+    recentProjectsMapper.setMapping(action, QString::fromStdString(filename));
+
+    // and add the new settings item
+    settings.beginWriteArray("recentProjects", size + 1);
+    settings.setArrayIndex(size);
+    settings.setValue("fileName", QString::fromStdString(filename));
+    settings.endArray();
+  }
```
Comment (jcoffland) [inline src/camotics/qt/QtWin.cpp]: I'd like to see this moved to it's own function.

File: qt/camotics.ui [hunk 3]
```diff
@@ -3042,6 +3049,11 @@
     <string>Reload this file</string>
    </property>
   </action>
+  <action name="actionFoobar">
+   <property name="text">
+    <string>Foobar</string>
```
Comment (jcoffland) [inline qt/camotics.ui]: What is the purpose of this property?

**Comment by martonmiklos** [comment]
I have added an important feature to the recent list: limited the history size to 20 items. Implementing this feature had eliminated the first review.

I have fixed the indentation issue, also removed the foobar action from the ui file. It was just used for testing, thanks for spotting it!

**Comment by jcoffland** [comment]
This looks great.  Thanks.  I'm going to commit a few minor formatting tweaks which you might want to pull before doing any further work.

**Comment by martonmiklos** [comment]
Cool! Thanks for the review and the heads up!

**Pull request merged**